### PR TITLE
update: migrate dp from sts to deployments

### PR DESCRIPTION
**Pull request by pratapaprasanna**

### CODE REVIEW

File: scripts/deploy_preview.sh [hunk 1]
```diff
@@ -28,14 +28,29 @@ export AWS_ACCESS_KEY_ID=$(echo $sts_output | jq -r '.Credentials''.AccessKeyId'
...
+then
+  kubectl delete ns $NAMESPACE || true
+  mongosh "mongodb+srv://$DB_USERNAME:$DB_PASSWORD@$DB_URL/$DBNAME?retryWrites=true&minPoolSize=1&maxPoolSize=10&maxIdleTimeMS=900000&authSource=admin" --eval 'db.dropDatabase()'
+  ACCESS_POINT_ID=$(aws efs describe-access-points --file-system-id "$DP_EFS_ID" | jq -r '.AccessPoints[] | select(.Name=="'"$PULL_REQUEST_NUMBER"'") | .AccessPointId')
+  aws efs delete-access-point --access-point-id $ACCESS_POINT_ID
+fi
+
+## Use DP-EFS and create ACCESS_POINT
+ACCESS_POINT=$(aws efs create-access-point --file-system-id $DP_EFS_ID --tags Key=Name,Value=$PULL_REQUEST_NUMBER)
+ACCESS_POINT_ID=$(echo $ACCESS_POINT | jq '.AccessPointId' | tr -d '"')
```
Comment (sharat87) [inline scripts/deploy_preview.sh:42]:
```suggestion
ACCESS_POINT_ID=$(echo $ACCESS_POINT | jq -r '.AccessPointId')
```

shot-2023-08-03-05-47-34

File: scripts/deploy_preview.sh [hunk 2]
```diff
@@ -59,18 +69,23 @@ kubectl create secret docker-registry $SECRET \
   --docker-username=$DOCKER_HUB_USERNAME \
   --docker-password=$DOCKER_HUB_ACCESS_TOKEN -n $NAMESPACE
 
-echo "Add appsmith-ce to helm repo"
-AWS_REGION=ap-south-1 helm repo add $HELMCHART $HELMCHART_URL
+echo "Add appsmith-ee to helm repo"
+AWS_REGION=us-east-2 helm repo add $HELMCHART $HELMCHART_URL
```
Comment (sharat87) [inline scripts/deploy_preview.sh:73]: Do we need the region for `helm repo add`?

```suggestion
helm repo add "$HELMCHART" "$HELMCHART_URL"
```

File: scripts/deploy_preview.sh [hunk 3]
```diff
@@ -59,18 +69,23 @@ kubectl create secret docker-registry $SECRET \
...
 
 echo "Deploy appsmith helm chart"
-helm upgrade -i $CHARTNAME appsmith/appsmith -n $NAMESPACE \
-  --create-namespace --recreate-pods --set image.repository=$DOCKER_HUB_ORGANIZATION/appsmith-dp --set image.tag=$IMAGE_HASH \
-  --set image.pullSecrets=$SECRET --set redis.enabled=false --set mongodb.enabled=false --set ingress.enabled=true \
+helm upgrade -i $CHARTNAME appsmith-ee/$HELMCHART -n $NAMESPACE --create-namespace --recreate-pods \
+  --set _image.repository=$DOCKER_HUB_ORGANIZATION/appsmith-dp --set _image.tag=$IMAGE_HASH \
```
Comment (sharat87) [inline scripts/deploy_preview.sh:77]: Tag comes from `image.tag` and not `_image.tag` no? Can you check once?

File: scripts/deploy_preview.sh (hunk 2 above)
Comment (pratapaprasanna) [reply scripts/deploy_preview.sh:73]: That is for a extra caution. However, i dont think we need it

File: scripts/deploy_preview.sh (hunk 3 above)
Comment (pratapaprasanna) [reply scripts/deploy_preview.sh:77]: I Think i confused you enough on `_image` 🥲 
_image.tag takes precedence and it should be _image.tag
You can check the example here 
https://github.com/appsmithorg/appsmith-ee/pull/1789/files#r1279090669
Case-3 in the comments

Comment (sharat87) [reply scripts/deploy_preview.sh:77]: Damn! This many variables made it so confusing... I think I'll bother you with my confusion forever though. 😭

Thanks for clarifying.

**Review by sharat87 — APPROVED** [review]

**Pull request merged**